gaphor/plugins/autolayout/tests/test_pydot.py

```diff
@@ -0,0 +1,109 @@
+import pytest
+
+from gaphor import UML
+from gaphor.core.modeling import Comment
+from gaphor.diagram.tests.fixtures import connect
+from gaphor.plugins.autolayout.pydot import AutoLayout, parse_edge_pos
+from gaphor.UML.diagramitems import (
+    ActionItem,
+    AssociationItem,
+    ClassItem,
+    CommentItem,
+    CommentLineItem,
+    ForkNodeItem,
+    GeneralizationItem,
+    InputPinItem,
+    ObjectFlowItem,
+    PackageItem,
+)
+
+
+def test_layout_diagram(diagram, create, event_manager):
+    superclass = create(ClassItem, UML.Class)
+    subclass = create(ClassItem, UML.Class)
+    gen = create(GeneralizationItem, UML.Generalization)
+    connect(gen, gen.tail, superclass)
+    connect(gen, gen.head, subclass)
+
+    auto_layout = AutoLayout(event_manager, None)
+    auto_layout.layout(diagram)
+
+    assert gen.head.pos != (0, 0)
+    assert gen.tail.pos != (0, 0)
+
+
+def test_layout_with_association(diagram, create, event_manager):
```

Comment:
Does this test, the comment one, and fork node one,  need an assert?

Reply:
The alternative is that an except would be raised. What I wanted to assert is that it does not crash ;). It's hard to make asserts about the positions the elements are located at after auto-layout.